runsc/cgroup/cgroup_v2.go

```diff
@@ -306,14 +315,22 @@ func (c *cgroupV2) MemoryLimit() (uint64, error) {
 	}
 	limStr = strings.TrimSpace(limStr)
 	if limStr == "max" {
-		return math.MaxUint64, nil
+		limStr, err := getValue(c.MakePath("../"), "memory.max")
+		if err != nil {
+			return 0, err
+		}
+		limStr = strings.TrimSpace(limStr)
+		if limStr == "max" {
+			return math.MaxUint64, nil
+		}
+		return strconv.ParseUint(limStr, 10, 64)
 	}
 	return strconv.ParseUint(limStr, 10, 64)
 }
 
 // MakePath builds a path to the given controller.
-func (c *cgroupV2) MakePath(controllerName string) string {
-	return filepath.Join(c.Mountpoint, c.Path)
+func (c *cgroupV2) MakePath(additionalPath string) string {
```

Comment:
Instead of modifying MakePath, could you just use `filepath.Dir(...)` on the result from MakePath?

Reply:
I've reverted this change.